### PR TITLE
Create maven.yml

### PR DESCRIPTION
**Pull request by jweiser**
I heard about GitHub Actions to setup continuous integration workflows.  I wanted to test it out.  This is the default configuration for Maven CI.

This could really help us with ensuring committed changes still pass unit tests with release and other repositories.

### CODE REVIEW

**Comment by jweiser** [comment]
> _Where_ does this CI run?

On GitHub's cloud environment using a Docker container.

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
```
Comment (SolomonShorser-OICR) [inline .github/workflows/maven.yml]: Why not specify a version? "latest" is variable but our server is not.

Comment (jweiser) [reply .github/workflows/maven.yml]: Fair point - done: https://github.com/reactome/release-data-exporter/pull/3/commits/9b9bbd3120e3cc1f88bcc5f6802018dc4f50980e

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# This workflow will build a Java project with Maven
+# For more information see: https://help.github.com/actions/language-and-framework-guides/building-and-testing-java-with-maven
+
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ develop ]
```
Comment (cookersjs) [inline .github/workflows/maven.yml]: Does this mean the presumed default branch is 'develop'? I think some EBI steps (perhaps all) have the default usable branch as 'master'

File: .github/workflows/maven.yml (hunk 1 above)
Comment (cookersjs) [reply .github/workflows/maven.yml]: And is 18.04 the version of our servers?

**Pull request merged**